### DISCUSSION TITLE
Airflow shows "Log Url" localhost:8080, although I am using airflow behind nginx proxy on hostname:443 port

### DISCUSSION THREAD

**Discussion by raulk89** [apache/airflow#21690] [Q&A]
**Apache Airflow version**: 1.10.11, 1.10.12

**Environment**: Virtual machine

- **OS** (e.g. from /etc/os-release): CentOS Linux release 7.8.2003 (Core)
- **Kernel** (e.g. `uname -a`): 3.10.0-1127.18.2.el7.x86_64
- **Install tools**: yum, pip3
- **Others**:

**What happened**: After executing DAG, I clicked on "running tasks", then opens up this page, where there is "Log Url" link. But this one does not open, since I am using it behind nginx proxy.

![image](https://user-images.githubusercontent.com/25365468/91230793-d4a32a00-e734-11ea-9b1d-3b0d0a80561f.png)

**What you expected to happen**:  The "Log Url" should be "https://hostname:443/..."

**How to reproduce it**:

**Just make sure that your ariflow webserver runs on http://localhost:8080, then**
`# yum install nginx`

**Add virtualhost conf and add the following:**
_You can probably get by with just using port 80 also, then you do not need ssl and certs_
_And then your application opens up http://hostname:80_

vi /etc/nginx/conf.d/airflow_webserver_proxy.conf
```
server {
 listen 80;
 server_name hostname.domain.com;
 return 301 https://$host$request_uri;
}

server {
 listen 443 ssl http2;
 server_name hostname.domain.com;

 location / {
  proxy_pass http://localhost:8080;
  proxy_set_header Host $host;
 }

 ssl_certificate /etc/pki/tls/certs/<<location to .cer file>>;
 ssl_certificate_key /etc/pki/tls/private/<<location to .key file>>;

 ssl_session_timeout 60m;
 ssl_dhparam /etc/ssl/certs/<<self signed cert>>;

 ssl_protocols TLSv1.2;
 add_header Strict-Transport-Security "max-age=31536000;" always;

}
```
`# systemctl start nginx`

And then open up web browser, start a random DAG, click on "running task" and then there should be this "Log Url"

**Anything else we need to know**: It happens always.

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by raulk89** [comment]
Well this :443 is omitted from the URL ofcourse, since it is https default port.
So URL is: 
`https://hostname.domain.com/...`

**Comment by mik-laj** [comment]
What is your value for `base_url` option in `webserver` section ?

**Comment by raulk89** [comment]
http://localhost:8080

I am using rbac also (airflow.cfg file rbac = True)

Haven't noticed that any other link is broken except this "Log Url".

Raul

**Comment by mik-laj** [comment]
@raulk89  it should be set to ``https://localhost:8080``.

**Comment by raulk89** [comment]
Why https..?
All other links are fine with http://localhost:8080

Raul

**Comment by JeroniMan** [comment]
@raulk89, hi, i have the same problem. Did you solve it? I try solutions from this issues, but it doesn't help me ..

**Comment by nuno-andre** [comment]
The problem is in the link's `href` attribute, which includes the base URL (`http://localhost:8080`).

```html
<td>
  <a href="http://localhost:8080/log?execution_date=2023-07-21T00..." data-original-title="" title="">
    <span class="material-icons" aria-hidden="true">reorder</span>
  </a>
</td>
```

A `href="/log?execution_date=2023-07-21T00..."` would solve this issue.

This URL is:
- built at https://github.com/apache/airflow/blob/d05e42e5d2081909c9c33de4bd4dfb759ac860c1/airflow/models/taskinstance.py#L765
- and displayed by https://github.com/apache/airflow/blob/d05e42e5d2081909c9c33de4bd4dfb759ac860c1/airflow/www/views.py#L5811

**Comment by csherrell** [comment]
Was there ever a solution to this? I am using Airflow 3.1.3 with an nginx proxy. The workers are all behind the proxy and the Airflow UI is external facing. So the log link in the email is http://airflow.internalserver.lan.:8080/ instead of https://example.com/airlfow/. The base_url of the workers is set to the internal link because they can not get outside the lan.
